util/set.go

```diff
@@ -0,0 +1,33 @@
+package util
+
+import (
+	"reflect"
+)
+
+type Set map[interface{}]bool
+
+func NewSet() *Set {
+	return new(Set)
+}
+
+func (set Set) Append(element interface{}) Set {
```

Comment:
I would split this method in two other methods, `AppendAll` to append elements from slices and `Append` single element.